File: code/lista4solver/exec9.py

""" 

9) Independente de consideracoes de precedencia, porque as declaracoes 
a = a++ e 
 i++ + ++i  sao de comportamento indefinido

""" 

def main():
	a = 2 
	i = 1

	#a = a++
	print a++ + ++a
	print i++ + ++i

main()

File: code/lista4solver/exec9.c
/***

9) Independente de considerações de precedência, porque as declarações 
“a = a++” e 
“i++ + ++i” são de comportamento indefinido?

***/
#include <stdio.h>
#include <stdlib.h>

int main(){
 int a;
 int i;
 
 a = a++;

 printf("a: %d\n", ++a);
 printf("i: %d\n",i++);
 printf("i: %d\n",++i);
 return 0;
}
